core/server/proxy/src/main/java/alluxio/proxy/s3/S3Handler.java

```diff
@@ -72,6 +75,13 @@ public class S3Handler {
           Pattern.compile("^" + S3RequestServlet.S3_V2_SERVICE_PATH_PREFIX + "/[^/]*$");
   public static final Pattern OBJECT_PATH_PATTERN =
           Pattern.compile("^" + S3RequestServlet.S3_V2_SERVICE_PATH_PREFIX + "/[^/]*/.*$");
+  public static final int BUCKET_PATH_CACHE_SIZE = 65536;
+  public static final Cache<AlluxioURI, Boolean> BUCKET_PATH_CACHE = CacheBuilder.newBuilder()
+      .maximumSize(BUCKET_PATH_CACHE_SIZE)
+      .expireAfterWrite(
+          Configuration.global().getMs(PropertyKey.PROXY_S3_BUCKETPATHCACHE_TIMEOUT_MS),
+          TimeUnit.MILLISECONDS)
+      .build();
```

Comment:
why keep boolean as value? if u delete the bucket why can't just remove it?

Reply:
also why keep alluxiouri as key? in S3RestUtils.checkPathIsAlluxioDirectory we anyways are using bucketPath to check right?

Reply:
> for the first question, `BUCKET_PATH_CACHE ` is a cache and there is a nearly perfect implement of cache in Alluxio while this cache is map type. So I set `BUCKET_PATH_CACHE ` as a map type as well. `boolean` is most suitable as the second key of this map.
 
isn't this guava cache instead of alluxio impl? 


> for the second question, ` S3RestUtils.checkPathIsAlluxioDirectory` calls master with arg `AlluxioURI(bucketPath)` to check. so `AlluxioURI(bucketPath)` is better than `bucketPath` to be the first key of `BUCKET_PATH_CACHE`.

but in every call the use of ` S3RestUtils.checkPathIsAlluxioDirectory` also uses bucketPath, so now its  1)conversion of bucketpath to alluxiouri 2) check cache with it 3) if not in cache checkPathIsAlluxioDirectory with bucketpath again and another conversion to alluxiouri and call master?  u see my point?  since from s3 api layer its already using string type bucketpath, why not cache the bucketpath str only?

Reply:
> > if u delete the bucket why can't just remove it?
> 
> if we introduce negative cache in the future, setting `false` will work different from removing it.

Actually here in current way of using cache in checkPathIsAlluxioDirectory, the cache is not treated as negative cache, AKA its not acting back directly from the cache if  1)cache hit and 2) value is negative, it still went ahead to check master instead of throwing bucket-not-exist exception. And regarding proxy should stay stateless at all times,  in order to prevent stale entires (e.g. delete bucket from this proxy but create same bucket on another proxy) , we should never use a negative cache here. So I would suggest :  1) keep the current way of check on master on cache miss  2) remove the boolean if u can find a way with guava cache or internal impl with a expiration,  3) or keep the boolean still but give it a comment of this boolean is simply a placeholder     Does that make sense to u?